src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +842,348 @@ def __try_add_error(error_list, detail):
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file, packages_retained_in_assessment, self.__assessment_total_error_count,
+                    self.__assessment_substatus_msg_copy, Constants.PATCH_ASSESSMENT_SUMMARY, assessment_substatus_index)
+
+            if len(installation_packages_removed_after_truncation) > 0:
+                # Update current installation # of removed packages
+                self.__installation_packages_removed = installation_packages_removed_after_truncation
+                # Recompose truncated status file payload (installation)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file, packages_retained_in_installation, self.__installation_total_error_count,
+                    self.__installation_substatus_msg_copy, Constants.PATCH_INSTALLATION_SUMMARY, installation_substatus_index)
+            status_file_size_in_bytes, status_file_agent_size_diff = self.__get_new_size_in_bytes_after_truncation(truncated_status_file)
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+        return truncated_status_file
+
+    def __get_new_size_in_bytes_after_truncation(self, truncated_status_file):
+        """ Get new size in bytes for status_file,and difference of status file - 126kb """
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(truncated_status_file)
+        status_file_agent_size_diff = status_file_size_in_bytes - self.__internal_file_capacity
+
+        return status_file_size_in_bytes, status_file_agent_size_diff
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_TRUNCATED_PACKAGE_COUNT
+        assessment_keep_five, assessment_packages_for_truncate = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
+            if len(assessment_packages) > min_packages_count else (assessment_packages, [])
+
+        return assessment_keep_five, assessment_packages_for_truncate
+
+    def __apply_truncation_process(self, assessment_packages, installation_packages, max_package_list_capacity, low_pri_index=None):
+        """ Truncation function call split assessment method and apply truncation on assessment and installation packages """
+        installation_low_pri = []
+        installation_high_pri = installation_packages
+        # cut assessment list into [:5], [5:]
+        assessment_keep_five, assessment_packages_for_truncate = self.__split_assessment_list(assessment_packages)
+
+        # Empty list after 2xjson.dumps have 4-5 bytes, might cause to lose 1 extra package in truncation process
```

Comment:
need more clarity in this comment

Reply:
good catch on this, I meant to move this comment to here. if I don't have that logic then the code highlighted in red will be executed and removing 2-4bytes from capacity, sometimes this possibly truncated 1 more package.

![image](https://github.com/Azure/LinuxPatchExtension/assets/127874208/fd13fb91-7782-43b3-a4e3-6f7a5afd34c0)